src/dbally/views/structured.py

```diff
@@ -57,20 +57,32 @@ async def ask(
             The result of the query.
         """
         iql_generator = self.get_iql_generator(llm)
-
         filters = self.list_filters()
         examples = self.list_few_shots()
+        aggregations = self.list_aggregations()
 
         iql = await iql_generator.generate_iql(
             question=query,
             filters=filters,
             examples=examples,
+            aggregations=[],
             event_tracker=event_tracker,
             llm_options=llm_options,
             n_retries=n_retries,
         )
         await self.apply_filters(iql)
 
+        iql = await iql_generator.generate_iql(
```

Comment:
Here we overwrite `iql`, so the context later is lacking information about the filters

Reply:
Done.